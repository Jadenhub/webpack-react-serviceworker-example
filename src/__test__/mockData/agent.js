export const MOBILE_AGENT = 'Mozilla/5.0 (iPhone; CPU iPhone OS 7_1_2 like Mac OS X) AppleWebKit/537.51.2 (KHTML, like Gecko) OPiOS/10.2.0.93022 Mobile/11D257 Safari/9537.53'
export const DESKTOP_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.246'