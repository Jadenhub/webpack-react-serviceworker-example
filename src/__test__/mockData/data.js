export const RATES = {
  "rates":{
    "btc": {
      "name":"Bitcoin",
      "unit":"BTC",
      "value":1.0,
      "type":"crypto"
    },
    "eth": {
      "name":"Ether",
      "unit":"ETH",
      "value":16.198,
      "type":"crypto"
    },
    "ltc": {
      "name":"Litecoin",
      "unit":"LTC",
      "value":230.349,
      "type":"crypto"
    },
    "bch": {
      "name":"Bitcoin Cash",
      "unit":"BCH",
      "value":64.588,
      "type":"crypto"
    },
    "bnb": {
      "name":"Binance Coin",
      "unit":"BNB",
      "value":105.539,
      "type":"crypto"
    },
    "eos": {
      "name":"EOS",
      "unit":"EOS",
      "value":8003.441,
      "type":"crypto"
    },
    "xrp": {
      "name":"XRP",
      "unit":"XRP",
      "value":47106.088,
      "type":"crypto"
    },
    "xlm": {
      "name":"Lumens",
      "unit":"XLM",
      "value":119836.321,
      "type":"crypto"
    },
    "link": {
      "name":"Chainlink",
      "unit":"LINK",
      "value":1693.022,
      "type":"crypto"
    },
    "dot":{
      "name":"Polkadot",
      "unit":"DOT",
      "value":1736.006,
      "type":"crypto"
    },
    "yfi":{
      "name":"Yearn.finance",
      "unit":"YFI",
      "value":0.985,
      "type":"crypto"
    }
  }
};
