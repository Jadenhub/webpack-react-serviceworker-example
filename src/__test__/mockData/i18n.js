export const I18N = {
  PROJECT_NAME: 'webapp',
  META_TITLE: 'Webapp',
  RATES: 'Rates',
  RATES_LIST: 'RATES LIST',
  NAME: 'Name',
  TYPE: 'Type',
  VALUE: 'Value',
  UNIT: 'Unit',
  SELECT_ALL: 'Select All',
  PAGE: 'Page',
  GO_TO_PAGE: 'Go to page',
  SHOW: 'Show',
  OF: 'of'
}