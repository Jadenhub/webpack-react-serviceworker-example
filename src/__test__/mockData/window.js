export function MockWindow(){
  global.window = {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn()
  }
}