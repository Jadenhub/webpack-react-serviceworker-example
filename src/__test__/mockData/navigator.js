import { DESKTOP_AGENT, MOBILE_AGENT } from "./agent";


export function MockDesktopNavigator(){
  global.navigator = {
    userAgent: DESKTOP_AGENT
  }
}

export function MockMobileNavigator(){
  global.navigator = {
    userAgent: MOBILE_AGENT
  }
}