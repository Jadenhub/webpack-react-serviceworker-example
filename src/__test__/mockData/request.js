import { DESKTOP_AGENT, MOBILE_AGENT } from "./agent"

export function getMockDesktopRequest(){
  return {
    path: '/',
    headers: {
      'user-agent': DESKTOP_AGENT
    }
  }
}

export function getMockMobileRequest(){
  return {
    path: '/',
    headers: {
      'user-agent': MOBILE_AGENT
    }
  }
}