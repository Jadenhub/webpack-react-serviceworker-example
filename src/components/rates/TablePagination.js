import { withI18n } from "../decorator/i18n";
import React from 'react';

function TablePagination({
  gotoPage,
  previousPage,
  nextPage,
  canPreviousPage,
  canNextPage,
  pageOptions,
  i18n,
  pageIndex,
  pageSize,
  setPageSize,
  pageCount
}){
  return (
    <div>
      <button onClick={() => gotoPage(0)} disabled={!canPreviousPage}>
        {'<<'}
      </button>{' '}
      <button onClick={() => previousPage()} disabled={!canPreviousPage}>
        {'<'}
      </button>{' '}
      <button onClick={() => nextPage()} disabled={!canNextPage}>
        {'>'}
      </button>{' '}
      <button onClick={() => gotoPage(pageCount - 1)} disabled={!canNextPage}>
        {'>>'}
      </button>{' '}
      <span>
        {i18n['PAGE']}{' '}
        <strong>
          {pageIndex + 1} {i18n['OF']} {pageOptions.length}
        </strong>{' '}
      </span>
      <span>
        | {i18n['GO_TO_PAGE']}:{' '}
        <input
          type="number"
          defaultValue={pageIndex + 1}
          onChange={e => {
            const page = e.target.value ? Number(e.target.value) - 1 : 0
            gotoPage(page)
          }}
          style={{ width: '100px' }}
        />
      </span>{' '}
      <select
        value={pageSize}
        onChange={e => {
          setPageSize(Number(e.target.value))
        }}
      >
        {[10, 20, 30, 40, 50].map(pageSize => (
          <option key={pageSize} value={pageSize}>
            {i18n['SHOW']} {pageSize}
          </option>
        ))}
      </select>
    </div>
  )
}


export default withI18n(TablePagination)