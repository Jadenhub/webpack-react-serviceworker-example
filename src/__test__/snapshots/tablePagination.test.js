import renderer from 'react-test-renderer';
import TablePagination from '../../components/rates/TablePagination';
import React from 'react';
import { I18N } from '../mockData/i18n';

test('renders correctly', () => {
  const tree = renderer
    .create(<TablePagination 
      gotoPage={jest.fn()}
      previousPage={jest.fn()}
      nextPage={jest.fn()}
      canPreviousPage={true}
      canNextPage={true}
      pageOptions={{
        length: 10
      }}
      i18n={I18N}
      pageIndex={0}
      pageSize={10}
      setPageSize={jest.fn()}
      pageCount={10}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});