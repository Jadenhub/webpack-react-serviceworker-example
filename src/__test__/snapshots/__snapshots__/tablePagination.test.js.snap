// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    disabled={false}
    onClick={[Function]}
  >
    &lt;&lt;
  </button>
   
  <button
    disabled={false}
    onClick={[Function]}
  >
    &lt;
  </button>
   
  <button
    disabled={false}
    onClick={[Function]}
  >
    &gt;
  </button>
   
  <button
    disabled={false}
    onClick={[Function]}
  >
    &gt;&gt;
  </button>
   
  <span>
    Page
     
    <strong>
      1
       
      of
       
      10
    </strong>
     
  </span>
  <span>
    | 
    Go to page
    :
     
    <input
      defaultValue={1}
      onChange={[Function]}
      style={
        Object {
          "width": "100px",
        }
      }
      type="number"
    />
  </span>
   
  <select
    onChange={[Function]}
    value={10}
  >
    <option
      value={10}
    >
      Show
       
      10
    </option>
    <option
      value={20}
    >
      Show
       
      20
    </option>
    <option
      value={30}
    >
      Show
       
      30
    </option>
    <option
      value={40}
    >
      Show
       
      40
    </option>
    <option
      value={50}
    >
      Show
       
      50
    </option>
  </select>
</div>
`;
